### PR TITLE
fix: ensure that the storage key does not alter the path structure

### PR DESCRIPTION
**Pull request by ctron**
The key has a direct impact on the path inside the S3 bucket. We must ensure that the user is not able to influence the location of the file, only the name of it.

This change ensures that they user provided key does not contain any character that might have an in impact on the location inside the bucket.

### CODE REVIEW

**Review by dejanb — APPROVED** [review]
LGTM

**Comment by ctron** [comment]
I had to re-implement that PR. The original idea, that we could limit the key character set, was flawed. In some cases we do use the CSAF document id as key, and we can't limit that value.

So I chose the alternative of URL-encoding the value instead.

File: integration-tests/tests/vexination.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ async fn vexination_roundtrip(vexination: &mut VexinationContext) {
     let id = input["document"]["tracking"]["id"].as_str().unwrap();
     let advisory = encode(id);
 
+    println!("ID: {id}");
```
Comment (dejanb) [inline integration-tests/tests/vexination.rs:22]: Are these printlns intentional?

Comment (ctron) [reply integration-tests/tests/vexination.rs]: Just removed them :)

**Review by dejanb — APPROVED** [review]

**Pull request merged**